include/dlaf/solver/triangular/impl.h

```diff
@@ -425,10 +425,48 @@ void Triangular<backend, device, T>::call_LLN(comm::CommunicatorGrid grid, blas:
   }
 }
 
-template <Backend backend, Device device, class T>
-void Triangular<backend, device, T>::call_LLT(comm::CommunicatorGrid grid, blas::Op op, blas::Diag diag,
-                                              T alpha, Matrix<const T, device>& mat_a,
-                                              Matrix<T, device>& mat_b) {
+template <Backend B>
+struct getGenericExecutor {
+  static auto call() {
+    return dlaf::getNpExecutor<Backend::MC>();
+  }
+};
+
+#ifdef DLAF_WITH_CUDA
+template <>
+struct getGenericExecutor<Backend::GPU> {
+  static auto call() {
+    return dlaf::cuda::Executor{dlaf::internal::getNpCudaStreamPool()};
+  }
+};
+#endif
+
+template <class T>
+void tileSubtractInPlace(matrix::Tile<T, Device::CPU> const& tile_a,
+                         matrix::Tile<const T, Device::CPU> const& tile_b) {
+  DLAF_ASSERT(equal_size(tile_a, tile_b), tile_a, tile_b);
+  for (auto j = 0; j < tile_a.size().cols(); ++j)
+    blas::axpy(tile_a.size().rows(), T(-1), tile_b.ptr({0, j}), 1, tile_a.ptr({0, j}), 1);
+}
+
+#ifdef DLAF_WITH_CUDA
+template <class T>
+void tileSubtractInPlace(cublasHandle_t handle, matrix::Tile<T, Device::GPU> const& tile_a,
+                         matrix::Tile<const T, Device::GPU> const& tile_b) {
+  DLAF_ASSERT(equal_size(tile_a, tile_b), tile_a, tile_b);
+  const T alpha = -1;
+  for (auto j = 0; j < tile_a.size().cols(); ++j)
+    tile::internal::CublasAxpy<T>::call(handle, tile_a.size().rows(), util::blasToCublasCast(&alpha),
+                                        util::blasToCublasCast(tile_b.ptr({0, j})), 1,
+                                        util::blasToCublasCast(tile_a.ptr({0, j})), 1);
+}
```

Comment:
Smart, I didn't think about batching operations together like this!

There's a potential tradeoff in that submitting each column seperately as a task (i.e. on a potentially separate stream) increases parallelism, but it also means paying the overhead of one future per column instead of one for the whole operation.

Question: do you know if there's a single batched operation that would do this in one blas/cublas call? I guess if the stride were the same as the row/column length you could call a single axpy with length rows * cols but for other strides it won't work.

Reply:
As pointed out by @rasolca, the best solution would be checking if MAGMA provides anything useful that operates in block, or just write a custom GPU kernel doing that.

This is some kind of workaround, let me know if it is acceptable for the moment.

Reply:
Fine by me at least, we'll have time to update it to something more performant in the future.